composer/datasets/synthetic.py

```diff
@@ -75,12 +63,22 @@ def __init__(self,
         self.memory_format = getattr(torch, MemoryFormat(memory_format).value)
         self.transform = transform
 
-        _validate_label_inputs(label_type=self.label_type, num_classes=self.num_classes, label_shape=self.label_shape)
+        self._validate_label_inputs(label_type=self.label_type,
+                                    num_classes=self.num_classes,
+                                    label_shape=self.label_shape)
 
         # The synthetic data
         self.input_data = None
         self.input_target = None
 
+    def _validate_label_inputs(self, label_type: SyntheticDataLabelType, num_classes: Optional[int],
+                               label_shape: Optional[Sequence[int]]):
+        if label_type == SyntheticDataLabelType.CLASSIFICATION_INT or label_type == SyntheticDataLabelType.CLASSIFICATION_ONE_HOT:
```

Comment:
skip the enums, and just match on strings.

Reply:
Didn't change the validation in this PR, but I would prefer to leave it as is. Since `SyntheticDataLabelType` is a `StringEnum`, it will perform a case-insensitive string comparision if `label_type` is a string. This is a private function (and yahp ensures that the yaml is parsed into the enum) so it should never be invoked with strings anyways.